### PR TITLE
Gradle 8 support / Misc fixes

### LINKED ISSUE #172: Add Gradle 8 support
The following error occurs when upgrading a project to Gradle 8. Most of other libraries have already upgraded, please upgrade, too. You are not forced to update to gradle 8, but at least you need to have a namespace in app/gradle, it's possible since Gradle 7.3

A problem occurred configuring project ':app_settings'.
> Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl.
   > Namespace not specified. Please specify a namespace in the module's build.gradle file
   
https://github.com/flutter/flutter/issues/125181

### PR DESCRIPTION
**Pull request by navaronbracke**
This PR updates the `app_settings` plugin to support Gradle 8 and it also fixes a few things I found along the way.

Changelog:
- Support Gradle 8, by adding the `namespace` attribute, per https://docs.flutter.dev/release/breaking-changes/android-java-gradle-migration-guide
- Regenerate the plugin with the correct organisation prefix, `com.spencerccf` instead of `com.example` (this was a guess)
- Remove the callback arguments, as an `await` is identical to what the callback did
- Fix the use of `result.success(null)` on Android
- Clean up the various different implementations of the public interface methods, now a single enum is used
- Support notification settings on iOS 16+

Fixes #172 

Apologies for the big diff, but most of it stems from the regeneration of the plugin (if only the `com.example` in the identifier was never a thing :/ ) Also, the iOS assets seem to have no rounded corners anymore (since iOS rounds the image anyway)

### CODE REVIEW

File: android/src/main/kotlin/com/spencerccf/app_settings/AppSettingsPlugin.kt [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "app_settings")
+    channel.setMethodCallHandler(this)
+  }
+
+  override fun onMethodCall(call: MethodCall, result: Result) {
+    val asAnotherTask = call.argument<Boolean>("asAnotherTask") ?: false
+
+    when(call.method) {
```
Comment (navaronbracke) [inline android/src/main/kotlin/com/spencerccf/app_settings/AppSettingsPlugin.kt:38]: FYI: Kotlin has a `when` operator that prevents having to use `if else`

File: android/src/main/kotlin/com/spencerccf/app_settings/AppSettingsPlugin.kt [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+      "accessibility" -> openSettings(Settings.ACTION_ACCESSIBILITY_SETTINGS, result, asAnotherTask)
+      "alarm" -> openAlarmSettings(result, asAnotherTask)
+      "apn" -> openSettings(Settings.ACTION_APN_SETTINGS, result, asAnotherTask)
+      "batteryOptimization" -> openBatteryOptimizationSettings(result, asAnotherTask)
+      "bluetooth" -> openSettings(Settings.ACTION_BLUETOOTH_SETTINGS, result, asAnotherTask)
```
Comment (navaronbracke) [inline android/src/main/kotlin/com/spencerccf/app_settings/AppSettingsPlugin.kt:43]: @spencerccf I tested the Bluetooth option on my old Android 10 phone and on an Android 13 emulator. Both worked without asking for permissions.

File: lib/app_settings_platform_interface.dart [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+import 'app_settings_method_channel.dart';
+import 'src/app_settings_type.dart';
+
+abstract class AppSettingsPlatform extends PlatformInterface {
```
Comment (navaronbracke) [inline lib/app_settings_platform_interface.dart:6]: @spencerccf The platform interface might not seem worthwhile, but it opens the door to adding other platforms in the future (for example if you ever want to integrate Windows or something)

**Comment by plinkos** [comment]
Could you resolve these merge conflicts?

**Comment by navaronbracke** [comment]
Yes, I'll look into it

**Comment by navaronbracke** [comment]
@plinkos Done! While I was at it, I also did the following:
- The readme for the example app was a copy of the toplevel one. This is not required.
- I cleaned up the Swift code since we now have _another_ case for the subscriptions

**Pull request merged**